cardano-api/src/Cardano/Api/Block.hs

```diff
@@ -14,24 +14,24 @@ module Cardano.Api.Block (
 
     -- * Blocks in the context of an era
     Block(.., Block),
-    BlockHeader,
+    BlockHeader(..), -- TODO is this the interface we really want to expose? Includes a `Hash` type family.
```

Comment:
Not sure about this one.

Reply:
Humm OK. As far as I remember, the use of Type Families was a source of unwanted complexity. Perhaps the rule of thumb for cardano-api is to avoid exporting anything with a type family unless exported from an "internal" module. This is a "internal" module, so we can export the constructors here, but not from `Cardano.Api`. Does that sound reasonable?

Reply:
On the other hand, the `Hash` type family is already exported from `Cardano.Api` so maybe we can just go along with that for now. Whether to export `Hash` or not now seems out of scope for this PR.